### PR TITLE
Utilize typed atomics

### PR DESCRIPTION
**Pull request by Stefan-Ethernal**
# Description

This PR utilizes typed atomics where appropriate, such as: `atomic.Bool` and `atomic.Pointer[T]`, which is possible now when we are running golang 1.19.

# Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [ ] Hotfix (change that solves an urgent issue, and requires immediate attention)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Checklist

- [ ] I have assigned this PR to myself
- [ ] I have added at least 1 reviewer
- [ ] I have added the relevant labels
- [ ] I have updated the official documentation
- [ ] I have added sufficient documentation in code

## Testing

- [ ] I have tested this code with the official test suite
- [ ] I have tested this code manually

### CODE REVIEW

File: consensus/polybft/consensus_runtime_test.go [hunk 1]
```diff
@@ -301,15 +301,15 @@ func TestConsensusRuntime_FSM_NotInValidatorSet(t *testing.T) {
 		Key: createTestKey(t),
 	}
 	runtime := &consensusRuntime{
-		proposerCalculator:  NewProposerCalculatorFromSnapshot(snapshot, config, hclog.NewNullLogger()),
-		activeValidatorFlag: 1,
```
Comment (Nemanja0x) [inline consensus/polybft/consensus_runtime_test.go:305]: Why are we missing this here and in all the other tests? True by default?

File: network/gossip.go [hunk 2]
```diff
@@ -126,7 +126,6 @@ func (s *Server) NewTopic(protoID string, obj proto.Message) (*Topic, error) {
 		topic:   topic,
 		typ:     reflect.TypeOf(obj).Elem(),
 		closeCh: make(chan struct{}),
-		closed:  new(uint64),
```
Comment (Nemanja0x) [inline network/gossip.go:129]: Default value?

File: network/gossip.go [hunk 3]
```diff
@@ -71,7 +71,7 @@ func (t *Topic) Subscribe(handler func(obj interface{}, from peer.ID)) error {
 	}
 
 	// Mark topic active.
-	atomic.StoreUint64(t.closed, 0)
+	t.closed.Swap(false)
```
Comment (Nemanja0x) [inline network/gossip.go]: Why do not use `Store()`

**Review by Nemanja0x — APPROVED** [review]
Please look at the comments. They are not blockers.

File: consensus/polybft/consensus_runtime_test.go (hunk 1 above)
Comment (Stefan-Ethernal) [reply consensus/polybft/consensus_runtime_test.go:305]: Nope, but I've set it to true here (seems that I cannot set it directly to true without need to call `setIsActiveValidator`, but maybe I'm missing out something):
`runtime.setIsActiveValidator(true)`

File: network/gossip.go (hunk 2 above)
Comment (Stefan-Ethernal) [reply network/gossip.go:129]: Fixed https://github.com/0xPolygon/polygon-edge/pull/1407/commits/09b94cecd1c6f00cc37961536058a8c6d0dbeed6. 

Although the documentation states that by default it is `false`.

File: network/gossip.go (hunk 3 above)
Comment (Stefan-Ethernal) [reply network/gossip.go]: Fixed https://github.com/0xPolygon/polygon-edge/pull/1407/commits/09b94cecd1c6f00cc37961536058a8c6d0dbeed6

**Review by igorcrevar — APPROVED** [review]

**Review by vcastellm — COMMENTED** [review]
Late LGTM

**Pull request merged**